lib/improver/blending/weighted_blend.py

```diff
@@ -811,25 +689,113 @@ def weighted_mean(self, cube, weights):
 
         Args:
             cube (iris.cube.Cube):
-                The cube which is being blended over self.coord.
+                The cube which is being blended over self.blend_coord.
             weights (iris.cube.Cube or None):
                 Cube of blending weights or None.
+
         Returns:
             iris.cube.Cube:
-                The cube with values blended over self.coord, with suitable
-                weightings applied.
+                The cube with values blended over self.blend_coord, with
+                suitable weightings applied.
         """
         weights_array = self.non_percentile_weights(cube, weights)
 
         # Calculate the weighted average.
-        cube_new = cube.collapsed(self.coord,
+        cube_new = cube.collapsed(self.blend_coord,
                                   iris.analysis.MEAN,
                                   weights=weights_array)
         cube_new.data = cube_new.data.astype(np.float32)
 
         return cube_new
 
-    def process(self, cube, weights=None):
+    @staticmethod
+    def _get_cycletime_point(input_cube, cycletime):
+        """
+        For cycle and model blending, establish the single forecast reference
+        time to set on the cube after blending.
+
+        Args:
+            input_cube (iris.cube.Cube):
+                Cube to be blended
+            cycletime (str or None):
+                The cycletime in a YYYYMMDDTHHMMZ format e.g. 20171122T0100Z.
+                If None, the latest forecast reference time is used.
+
+        Returns:
+            numpy.int64:
+                Forecast reference time point in units of input cube coordinate
+        """
+        frt_coord = input_cube.coord("forecast_reference_time")
+        if cycletime is None:
+            return np.max(frt_coord.points)
+        frt_units = frt_coord.units.origin
+        frt_calendar = frt_coord.units.calendar
+        cycletime_point = cycletime_to_number(
+            cycletime, time_unit=frt_units, calendar=frt_calendar)
+        return np.round(cycletime_point).astype(np.int64)
+
+    def _set_coords_to_remove(self, input_cube):
+        """
+        Generate a list of coordinate names associated with the blend
+        dimension.  Unless these are time-related coordinates, they should be
+        removed after blending.
+
+        Args:
+            input_cube (iris.cube.Cube):
+                Cube to be blended
+        """
+        time_coords = ["time", "forecast_reference_time", "forecast_period"]
+        blend_dim, = input_cube.coord_dims(self.blend_coord)
+        self.crds_to_remove = []
+        for coord in input_cube.coords():
+            if coord.name() in time_coords:
+                continue
+            if blend_dim in input_cube.coord_dims(coord):
+                self.crds_to_remove.append(coord.name())
+
+    def _set_forecast_reference_time_and_period(self, blended_cube):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method name "_set_forecast_reference_time_and_period" doesn't conform to '((test_[a-zA-Z0-9_]+)|([a-z_][a-z0-9_]{2,30}))$' pattern](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4464994)